### ISSUE TITLE
React Native 'ListView' display name selector not working with ShallowWrapper.find

### ISSUE THREAD

**Issue by akonwi** [airbnb/enzyme#827]
I have a react native component with a render method which looks like this:

``` jsx
    return (
      <View>
        <View>
          <TextInput/>
        </View>
        <ListView/>
      </View>
    )
```

In my test, `find('TextInput')` works correctly but when I try `find('ListView').exists()`, the result is incorrectly false. It works if I import `ListView` and call `find(ListView)` with the component constructor as the selector.

**Comment by brookslyrette** [comment]
Did you find a solution to this?

**Comment by akonwi** [comment]
no i didn't. Oddly enough using the string works in another test in the same file but it fails in another and that's where I use the function.

**Comment by ljharb** [comment]
In general, using string selectors is a code smell; you should use `.find(TextInput)` instead.

**Comment by akonwi** [comment]
How can string selectors be a code smell? Selectors are strings in regular DOM node searches. If enzyme prefers using the function over the name, then that should be documented and if it is, I haven't seen it.

**Comment by ljharb** [comment]
The DOM doesn’t provide you with anything better; enzyme does. (One way is that you might have two components named “TextInput”)

That enzyme supports finding by component is documented; that strings are always worse when other options exist is true in the general sense; perhaps we should add the recommendation to the docs - but it’s still just as true either way :-)

**Comment by akonwi** [comment]
[Enzyme docs](http://airbnb.io/enzyme/docs/api/selector.html#3-a-react-components-displayname) state that a component's display name can be used as a selector. I used a display name as a selector and it worked inconsistently, so there's a problem here and it could be faulty setup on my part or an issue with enzyme. I posted here for assistance to find out.

Saying that using a string selector is "generally worse" doesn't acknowledge the issue. If you truly believe that is the worst selector method of the four options on that page, then the documentation should state that. An even better step might be to stop supporting that feature if you're going to tell users it's a code smell and suggest they do something else.

**Comment by ljharb** [comment]
I totally agree that it should be working properly; if you could provide more background on your original problem, then this should probably be reopened?

**Comment by akonwi** [comment]
I haven't worked on the code where I had this issue in months and it's changed since the original post. I think that's part of why I closed the issue too. 

In the test though, I was doing a shallow render of my component, whose render method looks like this:

``` javascript
render() {
  return (
    <KeyboardAvoidingView>
      <View>
        <TextInput/>
      </View>
      <View>
        <ListView/>
      </View>
    </KeyboardAvoidingView>
  )
}
```

I'm not working on the codebase for this anymore so this isn't an issue for me anymore but if @brookslyrette has the same problem then it may be worth re-opening this issue.
